### ISSUE TITLE
declarationDir not used in extended tsconfig.json

### ISSUE THREAD

**Issue by SteveByerly** [ezolenko/rollup-plugin-typescript2#101]
#### What happens and why it is wrong
When using a `tsconfig.json` that extends another tsconfig configuraiton file, the `declarationDir` is ignored unless `"declaration": true` is set in the extended `tsconfig.json` specified in the `rollup.config`

Expected behavior:
The plugin respects `"declaration": true` in the base tsconfig file.

#### Environment 

do not believe this to be relevant

#### Versions
- typescript: `2.9.2`
- rollup: `0.63.4`
- rollup-plugin-typescript2: `0.16.1`

#### rollup.config.js

```javascript
// Lib
import path from 'path';
import * as pkg from './package.json';

// Plugins
import commonjs from 'rollup-plugin-commonjs';
import resolve from 'rollup-plugin-node-resolve';
import typescript from 'rollup-plugin-typescript2';

const moduleName = pkg.name;
const distDir = path.resolve(__dirname, '../../dist');

export default {
  input: path.resolve(__dirname, './src/index.ts'),
  output: {
    file: path.resolve(distDir, `${moduleName}.js`),
    format: 'cjs',
    sourcemap: true,
  },
  plugins: [
    resolve(),
    typescript({
      clean: true,
      include: [
        './src/**/*.ts',
      ],
      rollupCommonJSResolveHack: true,
      tsconfig: path.resolve(__dirname, './tsconfig.json'),
      useTsconfigDeclarationDir: true,
    }),
    commonjs(),
  ],
};

```

#### tsconfig.json

`tsconfig.json`
```javascript
{
  "extends": "../tsconfig.base",
  "compilerOptions": {
    "declarationDir": "../../dist/types",
    "outDir": "../../dist",
    "rootDir": "./src"
  },
  "exclude": [
    "test"
  ]
}
```

`tsconfig.base.json`
```javascript
{
  "formatCodeOptions": {
    "indentSize": 2,
    "tabSize": 2
  },
  "compilerOptions": {
    "target": "es5",
    "module": "es2015",
    "allowJs": false,
    "checkJs": false,
    "declaration": true,
    "declarationMap": true,
    "removeComments": false,
    "downlevelIteration": true,
    "strict": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
    "moduleResolution": "node",
    "esModuleInterop": true
  }
}
```

**Comment by SteveByerly** [comment]
I'd be more than happy to take the time and fix - wanted to verify the behavior first and/or if it is intentional/desired.

**Comment by ezolenko** [comment]
Looks like this is because rpt2 resets `declarationDir` if it doesn't detect `declaration` in raw tsconfig (before ts had a chance to parse it and merge things from tsconfig.base). 

PR is always welcome! Best way to fix it would be changing `getOptionsOverrides` to use parsed tsconfig (returned by `tsModule.parseJsonConfigFileContent`) and then apply generated overrides and parse tsconfig again before returning it from `parseTsConfig`. This way we'll get be most consistent with what tsc does on its own.

**Comment by SteveByerly** [comment]
great, thanks for the update. i'll do my best to get something going in the next week.

fortunately this isn't blocking anything - workaround is just to set the declaration in the extended tsconfig.

**Comment by DrSensor** [comment]
Any update on this? 🙂

**Comment by ezolenko** [comment]
Right, this was here for a while, should be fixed in master, will be in 17.1 sometime this week or so

**Comment by DrSensor** [comment]
Hi @ezolenko, is v17.1 has been released:question:
